GUI/MainChangeset.cs

```diff
@@ -22,22 +22,59 @@ public void UpdateChangesDialog(List<KeyValuePair<GUIMod, GUIModChangeType>> cha
                 return;
             }
 
+            m_Changeset = new List<ModChange>();
+            m_Changeset.AddRange(changeset.Where(change => change.ChangeType == GUIModChangeType.Remove));
+            m_Changeset.AddRange(changeset.Where(change => change.ChangeType == GUIModChangeType.Update));
+
+            IEnumerable<ModChange> left = changeset.Where(change => change.ChangeType == GUIModChangeType.Install);
+            CreateInstallList(left);
+
+            changeset = m_Changeset;
+
             foreach (var change in changeset)
             {
-                if (change.Value == GUIModChangeType.None)
+                if (change.ChangeType == GUIModChangeType.None)
                 {
                     continue;
                 }
 
-                var item = new ListViewItem {Text = String.Format("{0} {1}", change.Key.Name, change.Key.Version)};
+                var item = new ListViewItem {Text = String.Format("{0} {1}", change.Mod.Name, change.Mod.Version)};
 
-                var sub_change_type = new ListViewItem.ListViewSubItem {Text = change.Value.ToString()};
+                var sub_change_type = new ListViewItem.ListViewSubItem {Text = change.ChangeType.ToString()};
+
+                ListViewItem.ListViewSubItem description = new ListViewItem.ListViewSubItem();
+                description.Text = change.Reason.Reason.Trim();
+
+                if (change.ChangeType == GUIModChangeType.Update)
+                {
+                    description.Text = String.Format("Update to version {0}", change.Mod.LatestVersion);
+                }
 
                 item.SubItems.Add(sub_change_type);
+                item.SubItems.Add(description);
                 ChangesListView.Items.Add(item);
             }
         }
 
+        private void CreateInstallList(IEnumerable<ModChange> changes, ModChange parent=null)
```

Comment:
Any chance I can ask for method documentation on `CreateInstallList`? It's not immediately clear to me what's going on without reading the code. (And even then I find myself looking for docs to confirm my understanding.)